roles/jd-server/src/lib/core.rs

```diff
@@ -0,0 +1,183 @@
+                                mempool::error::handle_error(&err);
+                                handle_result!(sender_update_mempool, Err(err));
+                            }
+                        }
+                    }
+                    tokio::time::sleep(mempool_update_interval).await;
+                    // DO NOT REMOVE THIS LINE
+                    //let _transactions = mempool::JDsMempool::_get_transaction_list(mempool_cloned_.clone());
+                }
+            });
+
+            let mempool_cloned = mempool.clone();
+            let sender_submit_solution = sender.clone();
+            task::spawn(async move {
+                loop {
+                    let result = mempool::JDsMempool::on_submit(mempool_cloned.clone()).await;
+                    if let Err(err) = result {
+                        match err {
+                            JdsMempoolError::EmptyMempool => {
+                                if last_empty_mempool_warning.elapsed().as_secs() >= 60 {
+                                    warn!("{:?}", err);
+                                    warn!("Template Provider is running, but its mempool is empty (possible reasons: you're testing in testnet, signet, or regtest)");
+                                    last_empty_mempool_warning = std::time::Instant::now();
+                                }
+                            }
+                            _ => {
+                                // TODO here there should be a better error managmenet
+                                mempool::error::handle_error(&err);
+                                handle_result!(sender_submit_solution, Err(err));
+                            }
+                        }
+                    }
+                }
+            });
+        };
+
+        let cloned = config.clone();
+        let mempool_cloned = mempool.clone();
+        let (sender_add_txs_to_mempool, receiver_add_txs_to_mempool) = unbounded();
+        task::spawn(async move {
+            JobDeclarator::start(
+                cloned,
+                sender,
+                mempool_cloned,
+                new_block_sender,
+                sender_add_txs_to_mempool,
+            )
+            .await
+        });
+        task::spawn(async move {
+            loop {
+                if let Ok(add_transactions_to_mempool) = receiver_add_txs_to_mempool.recv().await {
+                    let mempool_cloned = mempool.clone();
+                    task::spawn(async move {
+                        match mempool::JDsMempool::add_tx_data_to_mempool(
+                            mempool_cloned,
+                            add_transactions_to_mempool,
+                        )
+                        .await
+                        {
+                            Ok(_) => (),
+                            Err(err) => {
+                                // TODO
+                                // here there should be a better error management
```

Comment:
Maybe not it's own issue but perhaps an overarching issue for tracking better error handling across the code base? Seems like this could be part of the refactoring documentation/story.
cc: @plebhash @pavlenex @rrybarczyk

Reply:
This will need to happen as part of a different PR really. Its good that we have more visibility now about the technical debt in the code though